docs/guidelines.md

```diff
@@ -0,0 +1,43 @@
+# Plugin Author Guidelines
+
+Netlify plugins listed in the plugins.json file of this repository can be installed directly within the Netlify UI from the plugins directory. To help ensure a consistent user experience, we've prepared the following guidelines for authors submitting plugins to the directory.
+
+Before submitting a plugin for the plugins directory, please read and follow the guidelines below.
+
+## Provide a zero-config default.
+
+Plugins with required `inputs` cannot be installed via the Netlify UI. Wherever possible, a plugin should include default options that allow the plugin to run without configuration. In cases where a plugin requires a unique value (such as an API key for a third-party service), configure the plugin to accept this value from a [build environment variable](https://docs.netlify.com/configure-builds/environment-variables). For more complex or customized configuration, users can install the plugin via the `netlify.toml` configuration file.
+
+## Provide a README.
+
+Every plugin in the Netlify plugins directory includes a link to the plugin README. This file should include:
+- A description of what the plugin does and why that might be useful.
+- Any required environment variables.
+- Instructions for installing via `netlify.toml` configuration file, including any optional `inputs`.
+- Ideally, a link to a demo site with [public deploy logs](https://docs.netlify.com/configure-builds/get-started/#basic-build-settings) and a [Deploy to Netlify button](https://docs.netlify.com/site-deploys/create-deploys/#deploy-to-netlify-button), so users can find out how the plugin works before installing.
+
+## Follow best practices for plugin code and metadata.
+
+Consistency across plugins makes plugins easier to find, easier to debug, and easier to review for inclusion in the Netlify plugins directory. This [issue comment](https://github.com/netlify/build/issues/1068#issuecomment-605276244) describes common pitfalls to avoid.
```

Comment:
Should we integrate the comment in its own Markdown document in this repository? It is still being updated, so leaving it as an issue comment of a separate repository might be odd, and does not allow for code review.

Reply:
I'm planning to add the contents of the comment to the main docs, and will update the link when available.

Regarding the portion about deprecate methods, I was wondering if we should keep a separate running list of which methods which have been deprecated or removed, with dates of deprecation and/or removal. I think this might make sense in the build repo. What do you think?

Reply:
Makes sense!